### ISSUE TITLE
torch.concatenate is not a proper alias for torch.cat in functorch

### ISSUE THREAD

**Issue by i404788** [pytorch/pytorch#91781]
### 🐛 Describe the bug

As stated the torch.concatenate is not aliased to torch.cat in functorch causing the following error: `RuntimeError: Batching rule not implemented for aten::concatenate. We could not generate a fallback.`

Reproduction:
```python3
import functorch
import torch

def cat(x, y):
    return torch.cat((x,y), dim=0)

def concatenate(x,y):
    return torch.concatenate((x,y), dim=0)

v = torch.rand(2,16)

# Works
print(functorch.vmap(cat)(v, v))

# RuntimeError
print(functorch.vmap(concatenate)(v, v))
```

### Versions

PyTorch version: 1.13.0
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: Void Linux (x86_64)
GCC version: (conda-forge gcc 10.4.0-16) 10.4.0
Clang version: 12.0.1
CMake version: version 3.25.1
Libc version: glibc-2.36

Python version: 3.10.6 | packaged by conda-forge | (main, Aug 22 2022, 20:35:26) [GCC 10.4.0] (64-bit runtime)
Python platform: Linux-6.0.15_1-x86_64-with-glibc2.36
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] adabelief-pytorch==0.2.1
[pip3] jax2torch==0.0.7
[pip3] lovely-numpy==0.2.8
[pip3] mypy-extensions==0.4.3
[pip3] numpy==1.23.3
[pip3] torch==1.13.0
[pip3] torchinfo==1.7.1
[pip3] torchvision==0.13.0a0+8069656
[conda] adabelief-pytorch         0.2.1                    pypi_0    pypi
[conda] jax2torch                 0.0.7                    pypi_0    pypi
[conda] lovely-numpy              0.2.8                    pypi_0    pypi
[conda] mkl                       2022.2.1         h84fe81f_16997    conda-forge
[conda] numpy                     1.23.3          py310h53a5b5f_0    conda-forge
[conda] pytorch                   1.13.0          cpu_py310h02c325b_0    conda-forge
[conda] torchinfo                 1.7.1              pyhd8ed1ab_0    conda-forge
[conda] torchvision               0.13.0          cpu_py310h9015461_0    conda-forge

cc @zou3519 @Chillee @samdow @soumith

**Comment by zou3519** [comment]
We would accept a PR fixing this. Should not be difficult (see https://github.com/pytorch/functorch/issues/1088 for examples)

**Comment by jasonkhadka** [comment]
I will take this.

**Comment by jasonkhadka** [comment]
@zou3519 This seems to be working in torch built from the latest master.

```python

import torch
import functorch

def cat(x, y):
    return torch.cat((x,y), dim=0)

def concatenate(x,y):
    return torch.concatenate((x,y), dim=0)

v = torch.rand(2,16)

print(functorch.vmap(cat)(v, v).shape)
# Output: torch.Size([2, 32])

print(functorch.vmap(concatenate)(v, v).shape)
# Output: torch.Size([2, 32])

print( torch.__version__)
# Output:  '2.0.0a0+git859ac58'
```

**Comment by i404788** [comment]
Hmm I did look into prior issues but it could be it was quitely added in dev

**Comment by princengare** [comment]
Hello,
Maybe changing torch.concatenate((x,y), dim=0) to torch.concat((x,y), dim=0) will fix this bug.

**Comment by anishlukk123** [comment]
what is the status of this issue, was this fixed, and if it isnt can it be assigned to me

**Comment by i404788** [comment]
According to @jasonkhadka It is resolved in 2.0.0 (alpha), so I'll close the issue for now.
